### PR TITLE
fix the issue 890 which throw exception Cannot destructure property '…

### PR DESCRIPTION
**Pull request by benzhuo**
…message' of 'tracker.fetch(...)' as it is null

### CODE REVIEW

**Comment by jsumners** [comment]
Thank you for contributing. We will need a test that fails as described in #890 without this change. My intention is to write an issue specific test that uses `ldapjs.createServer` to create a "server" that will induce the problem in the client. It would be similar to the test https://github.com/ldapjs/node-ldapjs/blob/f2890088e49c0c7b7b880998d73d6e4a448d7b4e/test/issue-845.test.js. I just haven't been able to dedicate the time to it yet.

**Comment by benzhuo** [comment]
I am not very understanding,  this is a very explicit syntax error.

**Comment by jsumners** [comment]
We need to include a unit test that covers the issue in order to guard against regression. I have provided guidance on how I intend to approach the test. I can provide further guidance if you have specific questions, or you can wait until I have time to get to it myself.

File: test/issue-890.test.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  return new Promise((resolve, reject) => {
+    t.context.client.destroy()
+    server.close((err) => {
+      if (err) return reject(err)
+      resolve()
+    })
+  })
+})
+
+tap.test('tracker.fetch', t => {
```
Comment (jsumners) [inline test/issue-890.test.js:61]: We can set a timeout for the test like `tap.test('handle null messages', { timeout: 2000 }, t => {`. Then use a lower `setTimeout` in the server to delay the response past the time the connecting client would expect it.

At least, that's my idea for how to write this test.

**Comment by jsumners** [comment]
Thank you for working on this @benzhuo. Unfortunately, the included test does not cover the crashing case. The primary work in this PR is developing that test. So I am closing this PR in favor of #913 which includes a working test.

**Pull request closed without merging**